src/config-factory.ts

```
@@ -27,3 +27,18 @@ export const configManager = createConfigManager();
+ * Simple event emitter for config changes.
+ */
+type ConfigChangeListener = () => void;
+const configChangeListeners: Set<ConfigChangeListener> = new Set();
+
+export function onConfigChanged(listener: ConfigChangeListener): () => void {
+  configChangeListeners.add(listener);
+  return () => configChangeListeners.delete(listener);
+}
+
+export function notifyConfigChanged(): void {
+  configChangeListeners.forEach((l) => l());
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)